subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -821,7 +821,7 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec {
         run "customTask", "--info"
         then:
         executedAndNotSkipped ":customTask"
-        output.contains "Value of input property 'bean.class' has changed for task ':customTask'"
+        output.contains "Value of input property 'bean' has changed for task ':customTask'"
```

Comment:
It would be nice to say that `Implementation of input property 'bean' has changed ...` here, and I guess we could do that easily by treating `ImplementationSnapshot` in a special way in `InputPropertyValueTaskStateChanges`, no?

Reply:
Better now?